### PR TITLE
Chore(emqx_management): merge enterprise mangement

### PR DESCRIPTION
**Pull request by Rory-Z**
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

## More information

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_api_pubsub.erl [hunk 1]
```diff
@@ -78,7 +78,19 @@ subscribe(_Bindings, Params) ->
 publish(_Bindings, Params) ->
     logger:debug("API publish Params:~p", [Params]),
     {ClientId, Topic, Qos, Retain, Payload} = parse_publish_params(Params),
-    return(do_publish(ClientId, Topic, Qos, Retain, Payload)).
+    case do_publish(ClientId, Topic, Qos, Retain, Payload) of
+        {ok, MsgIds} ->
+            case get_value(<<"return">>, Params, undefined) of
```
Comment (HJianBo) [inline apps/emqx_management/src/emqx_mgmt_api_pubsub.erl:83]: Need to update docs

File: apps/emqx_management/src/emqx_mgmt_cli.erl [hunk 2]
```diff
@@ -582,59 +543,21 @@ stop_listener(#{listen_on := ListenOn} = Listener, _Input) ->
...
-                        emqx_ctl:print("The emqx data import failed due: ~0p~n", [{Class,Reason,Stack}])
-                    end;
-                false ->
-                    emqx_ctl:print("Unsupported version: ~p~n", [Version])
-            end;
+    case emqx_mgmt_data_backup:import(Filename) of
+        ok ->
+            emqx_ctl:print("The emqx data has been imported successfully.~n");
+        {error, import_failed} -> 
+            emqx_ctl:print("The emqx data import failed. ~n");
```
Comment (HJianBo) [inline apps/emqx_management/src/emqx_mgmt_cli.erl:558]: Delete needless space

**Review by HJianBo — CHANGES_REQUESTED** [review]

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl [hunk 3]
```diff
@@ -0,0 +1,538 @@
...
+
+do_import_data(Data, Version) ->
+    import_blacklist(maps:get(<<"blacklist">>, Data, [])),
+    import_applications(maps:get(<<"apps">>, Data, [])),
+    import_users(maps:get(<<"users">>, Data, [])),
+    import_auth_clientid(maps:get(<<"auth_clientid">>, Data, [])),
+    import_auth_username(maps:get(<<"auth_username">>, Data, [])),
+    import_auth_mnesia(maps:get(<<"auth_mnesia">>, Data, []), Version),
+    import_acl_mnesia(maps:get(<<"acl_mnesia">>, Data, []), Version),
+    do_import_extra_data(Data, Version).
```
Comment (turtleDeng) [inline apps/emqx_management/src/emqx_mgmt_data_backup.erl:526]: do_import_extra_data Need first call

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl [hunk 4]
```diff
@@ -0,0 +1,538 @@
...
+
+-ifdef(EMQX_ENTERPRISE).
+do_import_extra_data(Data, Version) ->
+    import_confs(maps:get(<<"configs">>, Data, []), maps:get(<<"listeners_state">>, Data, [])),
+    import_resources(maps:get(<<"resources">>, Data, [])),
```
Comment (turtleDeng) [inline apps/emqx_management/src/emqx_mgmt_data_backup.erl:531]: import_resources_and_rules(maps:get(<<"resources">>, Data, []), maps:get(<<"rules">>, Data, []), Version).

**Review by turtleDeng — DISMISSED** [review]

**Review by HJianBo — DISMISSED** [review]

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl [hunk 5]
```diff
@@ -0,0 +1,540 @@
...
+    case ets:info(emqx_user) of
+        undefined -> ok;
+        _ ->
+            lists:foreach(fun(#{<<"username">> := Username, <<"password">> := Password}) ->
+                            mnesia:dirty_write({emqx_user, {username, Username}, base64:decode(Password), erlang:system_time(millisecond)})
+                          end, Lists)
+    end.
+
+import_auth_mnesia(Auths, FromVersion) when FromVersion =:= "4.0" orelse
+                                            FromVersion =:= "4.1" ->
```
Comment (tigercl) [inline apps/emqx_management/src/emqx_mgmt_data_backup.erl:344]: Add FromVersion =:=  "4.2" case

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+        _ ->
+            lists:foreach(fun(#{<<"login">> := Login,
+                                <<"type">> := Type,
+                                <<"password">> := Password,
+                                <<"created_at">> := CreatedAt }) ->
+                            mnesia:dirty_write({emqx_user, {any_to_atom(Type), Login}, base64:decode(Password), CreatedAt})
+                          end, Auths)
+    end.
+
+import_acl_mnesia(Acls, FromVersion) when FromVersion =:= "4.0" orelse
```
Comment (tigercl) [inline apps/emqx_management/src/emqx_mgmt_data_backup.erl:367]: Add FromVersion =:= "4.2" case

**Review by tigercl — DISMISSED** [review]

**Review by turtleDeng — APPROVED** [review]

**Pull request merged**